src/stream/src/executor/managed_state/top_n/top_n_state_new.rs

```diff
@@ -21,7 +21,7 @@ use risingwave_common::array::Row;
 use risingwave_common::catalog::{ColumnDesc, ColumnId, TableId};
```

Comment:
For managed state is there any situation where underlying keyspace is shared?
Or does managed state have exclusive access to it?
Asking because I'm wondering if there's a scenario where some executor using cell based encoding writes to managed state OR an executor using cell based decoding reads from it.

Reply:
Good catch! The only cases where we read and write in different paths are `DeltaJoinArrange`/`Lookup` and `Materialize`/`BatchQuery`. We should handle them carefully.